skiko/src/jvmMain/cpp/linux/drawlayer.cc

```diff
@@ -155,16 +159,36 @@ extern "C"
         dsi_x11 = (JAWT_X11DrawingSurfaceInfo *)dsi->platformInfo;
         Display *display = dsi_x11->display;
         Window window = dsi_x11->drawable;
+
+        LayerHandler *layer = new LayerHandler();
+        layerStorage->insert(layer);
+
+        jclass canvasClass = env->GetObjectClass(canvas);
+        static jmethodID renderApiMethod = NULL;
+        if (!renderApiMethod)
+        {
+            renderApiMethod = env->GetMethodID(canvasClass, "getApi", "()I");
+        }
+        if (NULL == renderApiMethod)

```

Comment:
Why not pass renderer API as an argument of `updateLayer` and do management on Kotlin side.